### PR TITLE
git: T5803: Adjust git configuration for baseline defaults

### PR DESCRIPTION
**Pull request by indrajitr**
## Change Summary

Apply baseline defaults for `.gitattributes` and `.vscode/settings.json` for improved developer experience.

The `.gitattrbutes` settings are based on:
- [Git documentation](https://git-scm.com/docs/gitattributes#_effects)
- [GitHub documentation](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings)
- [Community templates](https://github.com/gitattributes/gitattributes)

Since editor-agnostic line-ending specific settings are applied to `.gitattributes`, they can be removed from `.vscode/settings.json`. The global VSCode defaults have also been removed to avoid duplication.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe): git attributes, VSCode settings

## Related Task(s)

* https://vyos.dev/T5803

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,18 +1,26 @@
 {
-    "files.trimTrailingWhitespace": true,
-    "editor.tabSize": 4,
```
Comment (c-po) [inline .vscode/settings.json:3]: Why remove remove the whitespace indention setting?

File: .gitattributes [hunk 2]
```diff
@@ -1,4 +1,73 @@
...
+# *.svg    binary
+
+# Scripts
+*.bash     text eol=lf diff=bash
+*.sh       text eol=lf diff=bash
+
+# Serialisation
+*.json     text
+*.toml     text
+*.tmpl     text linguist-language=Jinja
```
Comment (c-po) [inline .gitattributes:44]: .j2 is missing for Jinja2

**Review by c-po — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (indrajitr) [reply .vscode/settings.json:3]: `"editor.insertSpaces": true` and `"editor.tabSize": 4` are VSCode defaults.
Besides, with `"editor.detectIndentation": true` (also VSCode default), `"editor.tabSize"` and `"editor.insertSpaces"` are automatically detected.

So they would be redundant.

Of course, one can make a case for the situation where a developer might have changed these settings in their local editor config. But in that case, we would have to apply project overrides for far too many settings to compensate for the developer's "creativity" with local settings 🙂. Such situations are usually rare though.

File: .gitattributes (hunk 2 above)
Comment (indrajitr) [reply .gitattributes:44]: Right you are, I missed `.j2` and some other extensions. Have added them and tweaked some other ones. Generally, line endings for most text files should be handled properly by git but good to have them explicitly listed.

**Review by c-po — APPROVED** [review]

**Comment by indrajitr** [comment]
_Note from https://vyos.dev/T5803 for somebody arriving here in the future from their custom fork:_

Since this PR affects the line-endings of files, once this PR is applied, one might need to reset the local git working tree from the local git index (after necessary care to `stash` or temporarily `commit` the local changes) by performing the following operations in the working directory:

```
git rm -rf --cached .
git reset --hard HEAD
```

**Comment by indrajitr** [comment]
@c-po can this be considered for sagitta so that Mergify doesn't fail for future backports on the affected files?

**Pull request merged**